client/client.go

```diff
@@ -89,4 +89,36 @@ type OpAMPClient interface {
 	// May be called before or after Start().
 	// May be also called from OnMessage handler.
 	RequestConnectionSettings(request *protobufs.ConnectionSettingsRequest) error
+
+	// SetCustomCapabilities modifies the set of customCapabilities supported by the client.
+	// The new customCapabilities will be sent with the next message to the server. SHOULD
+	// be called before Start(). If not called before Start(), the set of supported custom
+	// capabilities will be empty. May also be called anytime after Start(), including from
+	// OnMessage handler, to modify the set of supported custom capabilities. nil values are
+	// not allowed and will return an error.
+	//
+	// Each capability is a reverse FQDN with optional version information that uniquely
+	// identifies the custom capability and should match a capability specified in a
+	// supported CustomMessage. The client will automatically ignore any CustomMessage that
+	// contains a custom capability that is not specified in this field.
+	//
+	// See
+	// https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#customcapabilities
+	// for more details.
+	SetCustomCapabilities(customCapabilities *protobufs.CustomCapabilities) error
+
+	// SendCustomMessage sends the custom message to the Server. May be called anytime after
+	// Start(), including from OnMessage handler.
+	//
+	// If the CustomMessage is nil, ErrCustomMessageMissing will be returned. If the message
+	// specifies a capability that is not listed in the CustomCapabilities provided in the
+	// StartSettings for the client, ErrCustomCapabilityNotSupported will be returned.
+	//
+	// Only one message can be sent at a time. If SendCustomMessage has been already called
+	// and the message is still pending (in progress) then subsequent calls to
+	// SendCustomMessage will return ErrCustomMessagePending. To ensure that the previous
+	// send is complete and it is safe to send another CustomMessage, the caller should wait
+	// for the returned channel to be closed before attempting to send another custom
+	// message.
+	SendCustomMessage(message *protobufs.CustomMessage) (messageSendingChannel chan struct{}, err error)
```

Comment:
If an error is returned do we expect that messageSendingChannel is nil (or do we care?). Do you think we need to say anything about it or what we have is enough?

Reply:
If the error is `ErrCustomMessagePending`, then the messageSendingChannel is non-nil and the caller can wait on that channel before sending again. It is only currently nil if `ErrCustomMessageMissing` (nil message) or `ErrCustomCapabilityNotSupported` (unsupported message) are returned. I will tweak the comment to make that clear.

Reply:
Updated. Let me know if that is more clear.